### PR TITLE
Document [@error_message]

### PR DESCRIPTION
**Pull request by goldfirere**
As title. We somehow missed this.

@TheNumbat is a good writer and I would be pleased for his feedback here.

### CODE REVIEW

File: ocaml/jane/doc/extensions/error_message.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# The `[@error_message]` attribute
+
+You can put the `[@error_message]` attribute on type or kind constraints
+to add custom text to error messages. This can be useful in the output
```
Comment (TheNumbat) [inline ocaml/jane/doc/extensions/error_message.md:4]: Is this only intended for use in ppxes? If so it should say that explicitly

File: ocaml/jane/doc/extensions/error_message.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+of a ppx to better direct the user to the source of the error.
+
+For example, suppose we have a ppx that translates `[%i_have_a x]` to `Some x`.
+This makes sense only for `value`s. So we might actually translate
```
Comment (TheNumbat) [inline ocaml/jane/doc/extensions/error_message.md:8]: > so the ppx might actually translate

File: ocaml/jane/doc/extensions/error_message.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+let f x = [%i_have_a x]
+```
+
+to
+
+```
+let f x = Some (x : ((_ : value)[@error_message "only works with values"]))
+```
+
+which will say `only works with values` to the user in the error message.
```
Comment (TheNumbat) [inline ocaml/jane/doc/extensions/error_message.md:20]: > to the user if `x` does not have kind `value`.

**Review by TheNumbat — APPROVED** [review]

File: ocaml/jane/doc/extensions/error_message.md (hunk 1 above)
Comment (goldfirere) [reply ocaml/jane/doc/extensions/error_message.md:4]: I've connected the two sentences more -- see if you like.

**Pull request merged**